dvc/stage/cache.py

```diff
@@ -76,6 +77,45 @@ def _load(self, stage):
 
         return None
 
+    def _create_stage(self, cache):
+        from dvc.stage import create_stage, PipelineStage
+
+        stage = create_stage(
+            PipelineStage,
+            repo=self.repo,
+            path="dvc.yaml",
+            cmd=cache["cmd"],
+            deps=[dep["path"] for dep in cache["deps"]],
+            outs=[out["path"] for out in cache["outs"]],
+        )
+        StageLoader.fill_from_lock(stage, cache)
+        return stage
+
+    @contextmanager
+    def _cache_type_copy(self):
+        cache_types = self.repo.cache.local.cache_types
+        self.repo.cache.local.cache_types = ["copy"]
+        try:
+            yield
+        finally:
+            self.repo.cache.local.cache_types = cache_types
+
+    def _uncached_outs(self, stage, cache):
+        # NOTE: using temporary stage to avoid accidentally modifying original
+        # stage and to workaround `commit/checkout` not working for uncached
+        # outputs.
+        cached_stage = self._create_stage(cache)
+
+        outs_no_cache = [
+            out.def_path for out in stage.outs if not out.use_cache
+        ]
+
+        # NOTE: using copy link to make it look like a git-tracked file
+        with self._cache_type_copy():
```

Comment:
A bit weird that a method with such name does this implicitly. Will revisit in the morning on a fresh head.